webapp/src/assetEditor.tsx

```diff
@@ -25,10 +25,18 @@ export interface MessageData {
 
 export class AssetEditor extends React.Component {
     private editor: FieldEditorComponent<any>;
+    private currentJres: string;
 
     handleMessage = (msg: Message)  => {
-        if (msg.data._fromVscode && msg.data.type === "update") {
-            this.editor.loadJres(msg.data.message);
+        if (msg.data._fromVscode) {
+            if (msg.data.type === "initialize") {
+                this.editor.loadJres(msg.data.message);
+                this.currentJres = msg.data.message;
+            } else if (msg.data.type === "update") {
+                this.updateAndSendJres();
+            } else if (msg.data.type === 'updateWebview') {
+                this.editor.loadJres(this.currentJres);
```

Comment:
when do you send the updateWebview message? i'm not sure i see why you need to keep a separate "currentJres" variable

Reply:
I added this for the case when the user has the file open in two or more places in vscode like when they use the split editor. When they edit one and click done, the other open versions need to have the webview updated, but I'll remove this for now because I didn't finish implementing it on the vscode side and now that I'll thinking about it again, I think I should be able to take care of the issue using just the vscode side.